### DISCUSSION TITLE
Publish measured Voltage over MQTT

### DISCUSSION THREAD

**Discussion by tschaeffer2** [meatpiHQ/wican-fw#261] [Ideas]
Hello,
The WiCAN is a great tool. Thanks a lot!
There is one feature missing: The voltage measured by the device itself should be published as an MQTT message. 
Best regards,
Tobias

**Comment by meatpiHQ** [comment]
You can request the battery voltage by sending `{"cmd": "get_vbatt"}` to topic `wican/{device_id}/cmd`

**Reply by tschaeffer2** [reply]
Thanks a lot for the fast answer. I will try that out. Is there a reason why the information is not send cyclic as other common MQTT devices like TASMOTA, Shelly and ESPeasy are handling that?
And are there more internal commands (e.g. go to sleep mode, get status and get settings)?

**Reply by meatpiHQ** [reply]
@tschaeffer2 @njh I'll probably add the battery voltage publish in the next release. And I'll also make it available directly as a sensor in the HA integration.

The only other command available now is reboot.

**Reply by njh** [reply]
That would be great, thank you!

**Comment by njh** [comment]
I would love to have the battery voltage published periodically (along with CAN data), while connected to my home Wifi.
I can probably get Node-RED to send that command, when it sees WiCAN online but would be much easier if there was an option I could turn on.

**Comment by micha53** [comment]
I have a problem with the 12vBat data description with Zoe Ph2. 
In Home assistant I get an error Msg
```
state_topic: wican/7c2c67b3dbc1/auto
name: Zoe 12V-Batteriespannung
unit_of_measurement: V
value_template: '{{ value_json.12vBat}}'

Fehler: invalid template (TemplateSyntaxError: expected token 'end of print 
statement', got 'vBat') for dictionary value @ data['value_template'].

```
many other data is handled in the same way and is ok. Does anybody know, why I get this error msg? Maybe the 12vBat name is not allowed to begin with a number??

**Comment by snorre-k** [comment]
@meatpiHQ Also for me the periodically publishing of the battery data over MQTT would be a great help.

**Reply by meatpiHQ** [reply]
Do you mean the battery voltage?

**Reply by snorre-k** [reply]
@meatpiHQ Yes the 12v battery voltage seen by the Wican adapter

**Reply by meatpiHQ** [reply]
it is on the features request list
